### ISSUE TITLE
I get logged out of sites when re-opening Nyxt

### ISSUE THREAD

**Issue by anonimitoraf** [atlas-engineer/nyxt#1787]
**Precise recipe to reproduce the issue**
* Sign in to https://www.youtube.com or https://www.github.com
* Close Nyxt
* Open Nyxt
* Expectation: Still logged in | Reality: Logged out

Is there a setting I have to enable for persistent sessions?

For website-specific issues:
Can you reproduce this issue with Epiphany / GNOME Web (https://wiki.gnome.org/Apps/Web)?
No

**Information**
- OS name+version: Manjaro 21.1.0 Pahvo
- Graphics card and driver: 
NVIDIA GeForce GTX 1060 6GB
VGA compatible controller: NVIDIA Corporation GP106 [GeForce GTX 1060 6GB] (rev a1)
	Subsystem: Micro-Star International Co., Ltd. [MSI] Device 8c94
	Kernel driver in use: nvidia
	Kernel modules: nouveau, nvidia_drm, nvidia
- Desktop environment / Window manager name+version: i3 version 4.19.1 (2021-02-01) © 2009 Michael Stapelberg and contributors
- How you installed Nyxt (Guix pack, package manager, build from source): From source
- Information from `show-system-information`:
```
Nyxt version: 2.1.1-624-g935278d3
Renderer version: GI-GTK
Operating system kernel: Linux 5.12.19-1-MANJARO
Lisp implementation: SBCL 2.1.7 (Dynamic space size: 1073741824)
Features: (:WEBKIT2 :WEBKIT2-2.32 :WEBKIT2-PASTE-PLAINTEXT :WEBKIT2-TRACKING
 :WEBKIT2-MUTE :WEBKIT2-EMOJI :WEBKIT2-MEDIA :WEBKIT2-SANDBOXING :GTK-3-22
 :GTK-3-20 :GTK-3-18 :GTK-3-16 :GTK-3-14 :GTK-3-12 :GTK-3-10 :GTK-3-8 :GTK-3-6
 :GTK-3-4 :GTK :GDK-3-22 :GDK-3-20 :GDK-3-18 :GDK-3-16 :GDK-3-14 :GDK-3-12
 :GDK-3-10 :GDK-3-8 :GDK-3-6 :GDK-3-4 :CAIRO-1-10 :CAIRO-1-12 :GDK-PIXBUF
 :GLIB-2-30 :GLIB-2-32 :GLIB-2-34 :GLIB-2-36 :GLIB-2-38 :GLIB-2-40 :GLIB-2-42
 :GLIB-2-44 :GLIB-2-46 :GLIB-2-48 :GLIB-2-50 :GLIB-2-52 :GLIB-2-54 :GLIB-2-56
 :GLIB-2-58 :GLIB :NYXT-2 :FSET-EXT-STRINGS :CUSTOM-HASH-TABLE-NATIVE :SWANK
 :PLUMP-UTF-32 :CL-PPCRE-UNICODE :GLOBAL-VARS :DECLARE-TYPES :PARENSCRIPT
 :SBCL+SAFE-STANDARD-READTABLE :NAMED-READTABLES :LPARALLEL
 :LPARALLEL.WITH-CLTL2 :LPARALLEL.WITH-CAS :LPARALLEL.WITH-STEALING-SCHEDULER
 :21BIT-CHARS :CHUNGA CHIPZ-SYSTEM:GRAY-STREAMS :FAST-IO-SV :FAST-IO
 :SBCL-USES-SB-ROTATE-BYTE :ASDF-SYSTEM-CONNECTIONS :CLOSER-MOP :SPLIT-SEQUENCE
 :CL-UNICODE :FLEXI-STREAMS :CL-PPCRE :CL-JSON-DOUBLE-FLOAT-IS-SUBSUMED
 :CL-JSON-SINGLE-FLOAT-IS-SUBSUMED :CL-JSON-CLOS :CL-JSON :BORDEAUX-THREADS
 :THREAD-SUPPORT CFFI-FEATURES:FLAT-NAMESPACE CFFI-FEATURES:X86-64
 CFFI-FEATURES:UNIX :CFFI CFFI-SYS::FLAT-NAMESPACE ALEXANDRIA::SEQUENCE-EMPTYP
 :QUICKLISP :ASDF3.3 :ASDF3.2 :ASDF3.1 :ASDF3 :ASDF2 :ASDF :OS-UNIX
 :NON-BASE-CHARS-EXIST-P :ASDF-UNICODE :X86-64 :GENCGC :64-BIT :ANSI-CL
 :COMMON-LISP :ELF :IEEE-FLOATING-POINT :LINUX :LITTLE-ENDIAN
 :PACKAGE-LOCAL-NICKNAMES :SB-CORE-COMPRESSION :SB-LDB :SB-PACKAGE-LOCKS
 :SB-THREAD :SB-UNICODE :SBCL :UNIX)

ASDF version: 3.3.1
Critical dependencies: (/home/anonimito/common-lisp/nyxt/_build/submodules/cl-cffi-gtk/gtk/cl-cffi-gtk.asd
 /home/anonimito/common-lisp/nyxt/_build/quicklisp-client/dists/quicklisp/software/cl-gobject-introspection-20210124-git/cl-gobject-introspection.asd
 /home/anonimito/common-lisp/nyxt/_build/submodules/cl-webkit/webkit2/cl-webkit2.asd)

Quicklisp dist version: 2021-08-07
Quicklisp client version: 2020-01-04
Local project directories: (/home/anonimito/common-lisp/nyxt/_build/submodules/
 /home/anonimito/common-lisp/nyxt/_build/quicklisp-client/local-projects/)
Critical dependencies(#<SYSTEM cl-cffi-gtk / cl-cffi-gtk-20201220-git / quicklisp 2021-08-07>
 #<SYSTEM cl-gobject-introspection / cl-gobject-introspection-20210124-git / quicklisp 2021-08-07>
 #<SYSTEM cl-webkit2 / cl-webkit-20210807-git / quicklisp 2021-08-07>)

Guix version: 9875f9bca3976bf3576eab9be42164fde454597e
```

**Comment by jmercouris** [comment]
Could you please try with a more recent build? I believe this issue may have been at least /partially/ solved.

**Comment by anonimitoraf** [comment]
Hi @jmercouris 

I pulled `master` and rebuilt. I still get logged out of youtube :( but I don't seem to be logged out from github anymore.

What were the issues before that you think might be causing this?

**Comment by jmercouris** [comment]
I believe it to be an issue with how we are handling cookie and data caches. As per YouTube I suspect they are doing something sneaky which we are blocking, either on accident, or on purpose.

**Comment by anonimitoraf** [comment]
Ah, I see. How do I disable all the ad-block-like modes?

**Comment by anonimitoraf** [comment]
Ok, with this, I don't get logged out anymore:
```
(define-configuration (web-buffer nosave-buffer)
  ((default-modes '())))
```

So I guess it is one of the blocker-like modes

**Comment by anonimitoraf** [comment]
Actually, I've noticed that sometimes it logs me out. Sometimes it doesn't, intermittently.

I've disabled both `blocker-mode` and `reduce-tracking-mode`

This is my entire config:
```
(in-package :nyxt-user)

(defvar *swank-instance* (start-swank))

;; TODO Configure Emacs to fontify `def*'

;; Vim keybindings
(define-configuration buffer
  ((default-modes (append '(nyxt::vi-normal-mode) %slot-default%))))
(define-configuration prompt-buffer
  ((default-modes (append '(nyxt::vi-insert-mode) %slot-default%))))

;; Default modes
(define-configuration (web-buffer nosave-buffer)
  ((default-modes (append
                   '(auto-mode
                     ;; blocker-mode
                     ;; reduce-tracking-mode
                     ;; noimage-mode
                     ;; noscript-mode
                     ;; proxy-mode
                     force-https-mode)
                   %slot-default%))))
```

**Comment by Ambrevar** [comment]
This could be fully fixed since https://github.com/atlas-engineer/nyxt/commit/fc39a0d7f68030fd3cbe2da03fa9304648176280.
Can you people try `master` again?

**Comment by Mestelan** [comment]
Yes, it appears to be fixed here. Thank you!

**Comment by Mestelan** [comment]
Well, actually no, sorry: the issue still occurs (although much less frequently than before?). 
The issue tends to happen after Nyxt has exited in a crash.

**Comment by Ambrevar** [comment]
Fixed with #2050.
